lib/storage/src/content_manager/toc.rs

```diff
@@ -336,17 +344,31 @@ impl TableOfContent {
     }
 
     #[cfg(feature = "consensus")]
-    fn propose_collection_meta_op(
+    async fn propose_collection_meta_op(
         &self,
         operation: CollectionMetaOperations,
+        wait: bool,
     ) -> Result<bool, StorageError> {
-        match &self.propose_sender {
-            Some(sender) => sender.lock()?.send(serde_cbor::to_vec(&operation)?)?,
-            None => log::error!(
-                "Cannot submit collection meta operation proposal: no sender supplied to ToC"
-            ),
-        }
-        Ok(true)
+        let propose_sender = match &self.propose_sender {
+            Some(sender) => sender,
+            None => {
+                log::error!(
+                    "Cannot submit collection meta operation proposal: no sender supplied to ToC"
+                );
+                return Ok(true);
+            }
+        };
+        let serialized = serde_cbor::to_vec(&operation)?;
+        let result = if wait {
+            let (sender, receiver) = oneshot::channel();
+            self.on_meta_op_apply.lock()?.insert(operation, sender);
+            propose_sender.lock()?.send(serialized)?;
```

Comment:
Can `proposal` be rejected? Is it possible to track rejection?

Reply:
That's an interesting question, I will investigate. Theoretically if this peer is not a leader and it can't reach the leader for some time - it might be dropped, but I am not sure.

Also yes, you reminded me of necessity to introduce timeout there on commit wait!

Reply:
Would it be something like `?wait=10` for 10 seconds timeout?

Reply:
Yes, I think something like this would be good

Reply:
So I investigated some more:
1. It might drop proposal without returning error if leader is not elected (though raft will log this)
https://docs.rs/raft/latest/src/raft/raft.rs.html#2258-2269
2. If uncommitted limit is reached leader might drop proposal
3. If connection from peer to leader is unstable - proposal can be dropped by network - this case can be handled on our side with re-transmission.

It seems there is no general way to track the rejected entries in the raft library. I suggest that we use timeout for request wait, but mention in our docs that it might still be committed in some cases even if timeout for request wait passed.